app/main.py

```diff
@@ -39,51 +48,257 @@ def execute(self):
         """
 
+    @listen_to("kytos/topology.unloaded")
+    def unload_topology(self):  # pylint: disable=W0613
+        """Function meant for validation, to make sure that the shelve
+        has been loaded before all the other functions that use it begins to
+        call it."""
+        self.shelve_loaded = False  # pylint: disable=W0201
+
+    def test_kytos_topology(self):
+        """ Test if the Topology napp has loaded """
+        try:
+            _ = self.get_kytos_topology()
+            return True
+        except Exception as err:  # pylint: disable=W0703
+            log.debug(err)
+            return False
+
+    @staticmethod
+    def get_kytos_topology():
+        """retrieve topology from API"""
+        kytos_topology = requests.get(
+                settings.KYTOS_TOPOLOGY, timeout=10).json()
+        return kytos_topology["topology"]
+
+    def validate_sdx_topology(self):
+        """ return 200 if validated topology following the SDX data model"""
+        try:
+            response = requests.post(
+                    settings.SDX_TOPOLOGY_VALIDATE,
+                    json=self.sdx_topology,
+                    timeout=10)
+        except ValueError as exception:
+            log.info("validate topology result %s %s", exception, 401)
+            raise HTTPException(
+                    401,
+                    detail=f"Path is not valid: {exception}"
+                ) from exception
+        return {"result": response.json(), "status_code": response.status_code}
+
+    def convert_topology(self, event_type=0, event_timestamp=None):
+        """Function that will take care of update the shelve containing
+        the version control that will be updated every time a change is
+        detected in kytos topology, and return a new sdx topology"""
+        try:
+            # open the topology shelve
+            with shelve.open("topology_shelve") as open_shelve:
+                version = open_shelve['version']
+                self.dict_shelve = dict(open_shelve)  # pylint: disable=W0201
+                open_shelve.close()
+            if version >= 1 and event_type != 0:
+                timestamp = utils.get_timestamp()
+                if event_type == 1:
+                    version += 1
+                elif event_type == 2:
+                    timestamp = event_timestamp
+                return ParseConvertTopology(
+                    topology=self.get_kytos_topology(),
+                    version=version,
+                    timestamp=timestamp,
+                    model_version=self.dict_shelve['model_version'],
+                    oxp_name=self.dict_shelve['oxp_name'],
+                    oxp_url=self.dict_shelve['oxp_url'],
+                ).parse_convert_topology()
+            return {"result": topology_mock.topology_mock(),
+                    "status_code": 200}
+        except Exception as err:  # pylint: disable=W0703
+            log.info(err)
```

Comment:
In this case here, the correct log level would be `log.error`. Also, since you have a broad try/except block, it would be great to have more information on the context of the error to help troubleshooting when seeing the error on the log.

Reply:
Excellent, I will enhance the error description and modify it to log as an error.